app/views/teachers/TeacherStudentView.coffee

```diff
@@ -95,11 +79,122 @@ module.exports = class TeacherStudentView extends RootView
       $("[id|='visualisation']").hide()
       $(selected).show()
 
+  calculateStandardDev: ->
+    return unless @courses.loaded and @levels.loaded and @sessions?.loaded and @levelData
+
+    @courseComparisonMap = []
+    for versionedCourse in @classroom.get('courses') ? []
+      course = _.find @courses.models, (c) => c.id is versionedCourse._id
+      numbers = []
+      studentRate = 0
+      members = 0 #this is the COUNT for our standard deviation
+      for member in @classroom.get('members')
+        NUMBER = 0
+        memberBit = 0
+        for versionedLevel in versionedCourse.levels
+          for session in @sessions.models
+            if session.get('level').original == versionedLevel.original and session.get('creator') == member
+              # TODO IMPORTANT: only add number if @studentID in levelProgressMap has complete or started for the corresponding level
+              # in @levelData there's a levelProgress. If for this levelID, levelProgress = complete or started, go ahead.
+              temp = _.findWhere(@levelData, {levelID: session.get('level').original})
+              if temp.levelProgress == 'complete' or temp.levelProgress == 'started'
+                NUMBER += session.get('playtime') or 0
+                memberBit += 1
+              if session.get('creator') is @studentID
+                studentRate += session.get('playtime') or 0
+        if memberBit > 0 then members += 1
+        numbers.push NUMBER
+
+      # add all numbers[]
+      SUM = 0
+      for num in numbers
+        SUM += num
+
+      # divide by members to get MEAN, remember MEAN is only an average of the members' performance on levels THIS student has done.
+      MEAN = SUM/members
+
+      # make new list diffSquared[]
+      diffSquared = []
+
+      # for each number in numbers[], subtract MEAN then SQUARE, put new number in diffSquared
+      for num in numbers
+        diffSquared.push ((num-MEAN)*(num-MEAN))
+
+      # add all diffSquared[] then divide by COUNT to get VARIANCE
+      diffSum = 0
+      for num in diffSquared
+        diffSum += num
```

Comment:
Map and Reduce are some of my favorite functions ever, and make things like this so much more compact:
`diffSum = diffSquared.reduce((a,b) -> a+b, 0)`

See also https://derickbailey.com/2014/09/21/calculating-standard-deviation-with-array-map-and-array-reduce-in-javascript/

Lines 116-126 could be:
```
diffSum = numbers.map((num) -> (num-MEAN)**2).reduce((a,b) -> a+b, 0)
```